driver/pace/driver/state.py

```diff
@@ -58,3 +81,108 @@ class DriverState:
     grid_data: pace.util.grid.GridData
     damping_coefficients: pace.util.grid.DampingCoefficients
     driver_grid_data: pace.util.grid.DriverGridData
+
+    @classmethod
+    def load_state_from_restart(
```

Comment:
It would make sense to include a unit test that if you save a DriverState to restart and then load it, all data is the same (by iterating over every dataclass field and comparing). Would be helpful for diagnosing any issues that come up, e.g. if we add something to DriverState but we don't add it to the serialization.

Reply:
Added this test under `test_driver_state_load_restart`

Reply:
I don't see a test by that name, and the one test I do see calling `load_state_from_restart` doesn't test this (it compares the loaded state to the disk state, not to the before-saving state, so the conversion from before-saving to what's on disk is not tested).